java/src/jmri/ThrottleListener.java

```diff
@@ -38,67 +63,113 @@
+ *      ThrottleListener-> ThrottleManager : requestThrottle(LocoAddress,ThrottleListener,<b>true</b>)
+ *      ThrottleListener-> ThrottleManager : requestThrottle(BasicRosterEntry,ThrottleListener,<b>true</b>)
+ *  end
+ *  
+ *  group If Throttle Request Cannot Continue
+ *     ThrottleListener <-- ThrottleManager : return false 
+ *     deactivate ThrottleManager
+ *     note over ThrottleListener : Request ends here
+ *  end
+ *  deactivate ThrottleListener
+ * 
+ *  group If Throttle Request Can Continue
+ *     ThrottleListener <-- ThrottleManager : return true
+ *     note over ThrottleListener : Waits for callback to proceed.
+ *  end
+ *     
+ *  group If Throttle Object exists
+ *     
+ *      note over ThrottleManager : Sets BasicRosterEntry ( if present ) to Throttle
+ *      ThrottleListener <-- ThrottleManager: notifyThrottleFound(DccThrottle)
+ *      note over ThrottleListener : Throttle can now be controlled by ThrottleListener or a delegate.
+ *     
+ *  end
+ *     
+ *  group If Throttle Object does not exist
+ *     
+ *      note over ThrottleManager : Throttle Manager starts system specific actions to create a throttle.
+ *      note over ThrottleManager : If the throttle is available by normal means, no stealing or sharing,\nThrottle creation continues normally.
+ *      
+ *      == Optional: A steal or share decision is needed requestThrottle( .. , ThrottleListener, false) ==
  *
- * == Quick Failure Scenario ==
+ *      note over ThrottleManager : Abstract defaults to request steal, maintaining current behaviour.\nUsers prefs. may be queried\nThrottle creation continues normally.
+ *    
+ *      == Optional: A steal or share decision is needed requestThrottle( .. , ThrottleListener, true) ==
+ *       
+ *      note over ThrottleManager : User prefs. are not queried.
  *
- * ThrottleListener-> ThrottleManager : requestThrottle(address,ThrottleListener)
- * group If Throttle Request Cannot Continue
- *    ThrottleListener <-- ThrottleManager : return false 
- *    deactivate ThrottleManager
- *    note over ThrottleListener : Request ends here
- * end
- * deactivate ThrottleListener
+ *      ThrottleListener <-- ThrottleManager : notifyDecisionRequired(LocoAddress, ThrottleListener.DecisionType.STEAL)
+ *        
+ *      note over ThrottleListener : The question is to Steal or Cancel?
+ *        
+ *      ThrottleListener <-- ThrottleManager : notifyDecisionRequired(LocoAddress, ThrottleListener.DecisionType.SHARE)
+ *        
+ *      note over ThrottleListener : The question is to Share or Cancel?
+ *        
+ *      ThrottleListener <-- ThrottleManager : notifyDecisionRequired(LocoAddress, ThrottleListener.DecisionType.STEAL_OR_SHARE)
+ *        
+ *      note over ThrottleListener : The question is to Steal, Share or Cancel?
+ *        
+ *      note over ThrottleListener : The listener might query user prefs. to reach decision
  *
- * == Normal Request Scenario ==
```

Comment:
I think the new version of this diagram is rather unreadable, and it is in large part because some of the scenario markers were removed.

Reply:
Thanks for sorting this out